src/src_mysql/sadb_routine_mariadb.template.c

```diff
@@ -96,20 +99,111 @@ static int32_t sadb_config(void)
+    int32_t status = CRYPTO_LIB_ERROR;
+    if (NULL != sadb_mariadb_config) {
+        con = mysql_init(NULL);
+        //if encrypted connection (TLS) connection 
+        if (sadb_mariadb_config->encrypted_connection == 1) {
+            encrypted_tls_connection(con, sadb_mariadb_config);
+            //TLS needs no password 
+            //if NULL is returned then there is an error, else success
+            if (mysql_real_connect(con, sadb_mariadb_config->mysql_hostname,
+                    sadb_mariadb_config->mysql_username,
+                    sadb_mariadb_config->mysql_password,
+                    sadb_mariadb_config->mysql_database,
+                    sadb_mariadb_config->mysql_port, NULL, 0) == NULL) {
+                //0,NULL,0 are port number, unix socket, client flag
+                status = finish_with_error(con, SADB_MARIADB_CONNECTION_FAILED);
+            } else {
+                status = is_user_signed_into_db(con, sadb_mariadb_config->mysql_username);
+                if (status) {
+                    printf("sadb_init(), Using an encrypted connection \n");
+                }
+            }
+        }//end if TLS connection    
+            //else regular username & password connection 
+        else {
+            //if NULL is returned then there is an error, else success
+            if (mysql_real_connect(con, sadb_mariadb_config->mysql_hostname,
+                    sadb_mariadb_config->mysql_username,
+                    sadb_mariadb_config->mysql_password,
+                    sadb_mariadb_config->mysql_database,
+                    sadb_mariadb_config->mysql_port, NULL, 0) == NULL) {
+                //0,NULL,0 are port number, unix socket, client flag
+                status = finish_with_error(con, SADB_MARIADB_CONNECTION_FAILED);
+            } else {
+                status = CRYPTO_LIB_SUCCESS;
+            }
+        }//end regular password 
     }
-
+    return status;
+}
+/*===========================================================================
+Function:           encrypted_tls_connection
+Description:        Attempts to establish a encrypted connection using TLS.  
+Inputs:             MYSQL* con, SadbMariaDBConfig_t* sadb_mariadb_config         
+Outputs:            status - int32
+References:         1) https://dev.mysql.com/doc/c-api/8.0/en/c-api-encrypted-
+ *                      connections.html#c-api-enforcing-encrypted-connection
+ *                  2) https://dev.mysql.com/doc/c-api/8.0/en/mysql-ssl-set.html
+ *                  3) https://www.xuchao.org/docs/mysql/connectors-apis.html#c-api-encrypted-connections
+Example call:        
+Note:               MySQL server MUST be configured for encrypted connections:
+ *                  https://dev.mysql.com/doc/refman/5.7/en/using-encrypted-connections.html
+==========================================================*/
+static void encrypted_tls_connection(MYSQL* con, SadbMariaDBConfig_t* sadb_mariadb_config) {
+    if (NULL != con) {
```

Comment:
Lets' just remove this function and put the mysql_ssl_set function directly in the init function above.

Reply:
done.